include/dxc/Support/dxcapi.use.h

```diff
@@ -13,27 +13,40 @@
 #define __DXCAPI_USE_H__
 
 #include "dxc/dxcapi.h"
-#include "llvm/Support/DynamicLibrary.h"
 
 namespace dxc {
 
 // Helper class to dynamically load the dxcompiler or a compatible libraries.
 class DxcDllSupport {
 protected:
-  using DynamicLibrary = llvm::sys::DynamicLibrary;
-  DynamicLibrary m_dll;
+  HMODULE m_dll;
   DxcCreateInstanceProc m_createFn;
   DxcCreateInstance2Proc m_createFn2;
 
+  #ifndef _WIN32
+  void FreeLibrary(void* handle) {
+    ::dlclose(handle);
+  }
+  HMODULE LoadLibraryW(LPCWSTR name) {
+    char nameStr[256];
+    std::wcstombs(nameStr, name, 256);
+    return ::dlopen(nameStr, RTLD_LAZY);
+  }
+  HMODULE GetProcAddress(HMODULE dll, LPCSTR fnName) {
```

Comment:
Return type for the win32 version is FARPROC (which is defined: `typedef INT_PTR (FAR WINAPI *FARPROC)();` ).  I don't think it creates any opportunity for a bug here (since HMODULE is essentially a pointer), but it does look odd that its return type is HMODULE when it returns a function pointer.